roles/consul/defaults/main.yml

```diff
@@ -5,7 +5,8 @@ consul_dc: dc1
 consul_servers_group: consul_servers
 consul_advertise: "{{ ansible_default_ipv4.address }}"
 consul_bind_addr: "{{ ansible_default_ipv4.address }}"
-consul_bootstrap_expect: "{{ groups[consul_servers_group] | length }}"
+consul_retry_join: "{% for host in groups[consul_servers_group] | intersect(groups[consul_dc]) %}\"{{ hostvars[host].ansible_default_ipv4.address }}\"{% if not loop.last %}, {% endif %}{% endfor %}"
```

Comment:
consul was failing to elect a leader and then failing to show up as services in the UI. Adding this fixed it.
-retry-join - Similar to -join but allows retrying a join if the first attempt fails. This is useful for cases where we know the address will become available eventually.